dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/ITextCompletion.cs

```diff
@@ -12,25 +12,26 @@ namespace Microsoft.SemanticKernel.AI.TextCompletion;
 public interface ITextCompletion
 {
     /// <summary>
-    /// Creates a completion for the prompt and settings.
+    /// Get completion results for the prompt and settings.
     /// </summary>
     /// <param name="text">The prompt to complete.</param>
     /// <param name="requestSettings">Request settings for the completion API</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    /// <returns>Text generated by the remote model</returns>
-    public Task<string> CompleteAsync(
+    /// <returns>List of different completions results generated by the remote model</returns>
+    public Task<IReadOnlyList<ITextCompletionResult>> GetCompletionsAsync(
         string text,
         CompleteRequestSettings requestSettings,
         CancellationToken cancellationToken = default);
 
     /// <summary>
-    /// Creates a completion for the prompt and settings.
+    /// Get streaming completion results for the prompt and settings.
     /// </summary>
     /// <param name="text">The prompt to complete.</param>
     /// <param name="requestSettings">Request settings for the completion API</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
-    /// <returns>Streaming content of the text generated by the remote model</returns>
-    public IAsyncEnumerable<string> CompleteStreamAsync(string text,
+    /// <returns>List of different completion streaming results generated by the remote model</returns>
+    public IAsyncEnumerable<ITextCompletionStreamingResult> GetStreamingCompletionsAsync(
```

Comment:
Note that the "public" here and above are ok but not actually needed, since this is an interface.